src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java

```diff
@@ -0,0 +1,193 @@
+					"org.apache.hadoop.fs.s3a.S3A");
+		} catch(Exception e) {
+			// lets not do AWS
+		}
+				
+		Assertions.assertNotNull(spark);
+	}
+
+	@AfterEach
+	public void after() {
+		spark.stop();
+		// cleanup derby database
+	}
+	
+	// Test
+	protected boolean hasField(final Dataset<Row> in, final String fieldname) {
+		return Arrays.asList(in.schema().fieldNames()).contains(fieldname);
+	}
+	
+	protected Object getFirstValue(final Dataset<Row> in, final String fieldname) {
+		final Row first = in.first();
+		return first.get(first.fieldIndex(fieldname));
+	}
+	
+	protected Dataset<Row> getPayload(Dataset<Row> df, final String column, final String parsed) {
+		final DataType schema = getSchema(df, column);
+		return df.withColumn(parsed, from_json(col(column), schema));
+	}
+
+	protected DataType getSchema(Dataset<Row> df, final String column) {
+		final Row[] schs = (Row[])df.sqlContext().range(1).select(
+				schema_of_json(lit(df.select(column).first().getString(0)))
+		).collect();
+		final String schemaStr = schs[0].getString(0);
+
+		return DataType.fromDDL(schemaStr);
+	}
+	
+	
+	protected Dataset<Row> loadDataframe(final String resource, final String filename) throws IOException {
+		return spark.read().load(createFileFromResource(resource, filename).toString());
+	}
+	
+	protected Dataset<Row> loadCsvDataframe(final String resource, final String filename) throws IOException {
+		return spark.read().option("header", "true").format("csv").load(createFileFromResource(resource, filename).toString());
+	}
+	
+	protected Dataset<Row> loadJsonDataframe(final String resource, final String filename) throws IOException {
+		return spark.read().option("multiline", "true").format("json").load(createFileFromResource(resource, filename).toString());
+	}
+	
+	protected Dataset<Row> loadParquetDataframe(final String resource, final String filename) throws IOException {
+		return spark.read().parquet(createFileFromResource(resource, filename).toString());
+	}
+
+	protected Path createFileFromResource(final String resource, final String filename) throws IOException {
+		final InputStream stream = getStream(resource);
+		final File f = Paths.get(filename).toFile();
+		FileUtils.copyInputStreamToFile(stream, f);
+		return Paths.get(f.getAbsolutePath());
+	}
+
+	protected Path createFileFromResource(final String resource, final String filename, final String directory) throws IOException {
+		final InputStream stream = getStream(resource);
+		File dir;
+		if(Files.exists(Paths.get(folder.toFile().getAbsolutePath(),directory))) {
+			dir = new File(folder.toFile().getAbsolutePath() + "/" + directory);
+		} else {
+			dir = Paths.get(directory).toFile();
+		}
+		final File f = new File(dir, filename);
+		FileUtils.copyInputStreamToFile(stream, f);
+		return Paths.get(f.getAbsolutePath());
+	}
+
+	@SuppressWarnings("deprecation")
+	protected String getResource(final String resource) throws IOException {
```

Comment:
Take a look at the code that reads resource files in the SourceReferenceService. Consider duplicating that in some sort of test helper (the code here is simpler and will be removed soon since it's temporary).